src/helpers/contacts.js

```diff
@@ -146,6 +146,34 @@ export const reworkContacts = (
   return reworkedContacts
 }
 
+const cleanFormattedAddress = formattedAddress => {
+  // Replace all spaces by one space, to fix cases where there are multiple spaces
+  // Replace commas that have a space before
+  // And remove all spaces before & after the string
+  let formattedAddressClean = formattedAddress
+    .replace(/\s+/g, ' ')
+    .replace(/\s,/g, '')
+    .trim()
+
+  // Case where a comma is the last character
+  if (
+    formattedAddressClean.lastIndexOf(',') ===
+    formattedAddressClean.length - 1
```

Comment:
super nit: `formattedAddressClean[formattedAddressClean.length - 1] === ','` is maybe clearer